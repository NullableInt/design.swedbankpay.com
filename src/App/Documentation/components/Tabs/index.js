import React, { Component } from "react";

import { ComponentPreview, DocToc, ExperimentalComponentAlert } from "#";
import { tabs } from "$/px-script";
import TabsComponent from "@/Tabs";

const tabItems = ["Kort", "Rabatter", "Transaksjoner", "Faktura", "Instillinger", "Audit trail"];

const Overview = () => (
    <>
        <h2 id="overview">Overview</h2>
        <p>Overview... <code className="token property">.tabs</code>...<code className="token property">.tabs-auto</code></p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TabsComponent items={tabItems} id="horizontal-list-demo"/>
        </ComponentPreview>
    </>
);

const Collapsed = () => (
    <>
        <h2 id="collapsed">Responsive</h2>
        <p>Responsive... <code className="token property">.tabs</code>...<code className="token property">.tabs-collapsed</code></p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <TabsComponent mode=" tabs-horizontal" items={tabItems} id="responsive-list-demo"/>
        </ComponentPreview>
    </>
);

const TabsText = () => (
    <div className="col-md-12 col-lg-10 doc-body">
        <ExperimentalComponentAlert />
        <p className="lead">Tabs...</p>
        <Overview />
        <Collapsed />
    </div>
);

class Tabs extends Component {
    componentDidMount () {
        tabs.init();
    }

    render () {
        return (
            <div className="doc-container">
                <div className="row">
                    <TabsText />
                    <DocToc component={TabsText} />
                </div>
            </div>
        );
    }
}

export default Tabs;

/* For testing */
export { Overview, Collapsed, TabsText };
