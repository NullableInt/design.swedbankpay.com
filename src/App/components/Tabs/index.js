import React, { Component } from "react";
import PropTypes from "prop-types";

class Tabs extends Component {
    constructor (props) {
        super(props);

        this.state = { active: 1 };
    }

    setActive (e, i) {
        e.preventDefault();
        this.setState({ active: i });
    }

    render () {
        const { mode, items } = this.props;

        return (
            <div className="tabs">{"\n"}
                {mode === "responsive" ? <i className="material-icons">keyboard_arrow_right</i> : null}
                <ul className={`tabs-${mode}`}>
                    {items.map((name, i) => (
                        <li key={`tab-item-${name}-${i}`} className={this.state.active === i ? "active" : null}>{"\n"}
                            <a href="#" onClick={e => this.setActive(e, i)}>{name}</a>{"\n"}
                        </li>
                    ))}
                </ul>
            </div>
        );
    }
}

Tabs.propTypes = {
    mode: PropTypes.oneOf(["horizontal", "vertical", "responsive"]).isRequired,
    items: PropTypes.array.isRequired
};

export default Tabs;
