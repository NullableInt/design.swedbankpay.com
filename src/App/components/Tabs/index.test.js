import React from "react";
import { shallow } from "enzyme";

import Tabs from "./index";

const items = ["test", "test 2"];

// TODO: Write more tests!! [EH]

describe("Component: Tabs - ", () => {
    it("is defined", () => {
        expect(Tabs).toBeDefined();
    });

    it("Prop mode is required", () => {
        console.error = jest.fn();

        const wrapper = shallow(<Tabs mode="auto" items={items} />);

        expect(console.error).toHaveBeenCalled();
        expect(wrapper).toMatchSnapshot();
    });

    it("renders", () => {
        const wrapper = shallow(<Tabs items={items} />);

        expect(wrapper).toMatchSnapshot();
    });
});
