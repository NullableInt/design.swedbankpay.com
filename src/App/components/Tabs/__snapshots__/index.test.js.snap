// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Tabs -  Prop mode is required 1`] = `
<div
  className="tabs tabs-horizontal-auto"
>
  

  <i
    className="material-icons"
  >
    keyboard_arrow_right
  </i>
  <ul>
    <li
      className={null}
      key="tab-item-test-0"
    >
      

      <a
        href="#"
        onClick={[Function]}
      >
        test
      </a>
      

    </li>
    <li
      className="active"
      key="tab-item-test 2-1"
    >
      

      <a
        href="#"
        onClick={[Function]}
      >
        test 2
      </a>
      

    </li>
  </ul>
</div>
`;

exports[`Component: Tabs -  renders 1`] = `
<div
  className="tabs"
>
  

  <i
    className="material-icons"
  >
    keyboard_arrow_right
  </i>
  <ul>
    <li
      className={null}
      key="tab-item-test-0"
    >
      

      <a
        href="#"
        onClick={[Function]}
      >
        test
      </a>
      

    </li>
    <li
      className="active"
      key="tab-item-test 2-1"
    >
      

      <a
        href="#"
        onClick={[Function]}
      >
        test 2
      </a>
      

    </li>
  </ul>
</div>
`;
